src/odb/src/db/dbSite.cpp

```diff
@@ -230,6 +232,62 @@ void dbSite::setClass(dbSiteClass type)
   site->_flags._class = type.getValue();
 }
 
+void dbSite::setRowPattern(
+    std::vector<std::pair<std::string, std::string>>& row_pattern)
+{
+  _dbSite* site = (_dbSite*) this;
+  site->_row_patterns.reserve(row_pattern.size());
+  site->_flags._is_hybrid = 1;
+  std::unordered_map<std::string, dbOrientType::Value> orientationMap
+      = {{"N", dbOrientType::R0},
+         {"W", dbOrientType::R270},
+         {"S", dbOrientType::R180},
+         {"E", dbOrientType::R90},
+         {"FN", dbOrientType::MYR90},
+         {"FW", dbOrientType::MY},
+         {"FS", dbOrientType::MXR90},
+         {"FE", dbOrientType::MX}};
+
+  for (auto& row : row_pattern) {
+    dbOrientType orient = dbOrientType::R0;
+    auto _site = (_dbSite*) getLib()->findSite(row.first.c_str());
```

Comment:
You should check the site is found and error if it is not found.

Reply:
The same comment below applies here. 
please let me know if you think otherwise, but this was intentional from me